src/rascal/representations/calculator_spherical_invariants.hh

```diff
@@ -550,6 +547,10 @@ namespace rascal {
                 continue;
               }
 
+              auto && soap_neigh_gradient_by_species_pair{
```

Comment:
Why the `auto &&`here? Could `const auto&` do the job?

Reply:
I think nothing forbids us to use the auto && in the coding convention so I use it instead of something else. Why would you not use this language feature ?

Reply:
As always, because I like to see the types, and `auto &&` breaks that, since it can bind by value or by reference.